src/Silverback.Integration/Messaging/Inbound/ErrorHandling/MoveMessageErrorPolicy.cs

```diff
@@ -139,15 +142,21 @@ public override bool CanHandle(ConsumerPipelineContext context, Exception except
 
             private async Task PublishToNewEndpointAsync(IRawInboundEnvelope envelope, Exception exception)
             {
-                envelope.Headers.AddOrReplace(DefaultMessageHeaders.SourceEndpoint, envelope.ActualEndpointName);
+                MessageHeaderCollection envelopeHeaders = envelope.Headers;
```

Comment:
It might be a detail but you are again modifying the headers in the inbound envelope.
I really think that you should clone this collection (you can skip the timestamp and other unwanted headers on the fly).

It's wrong that it wasn't cloned in the previous implementation.

Reply:
I changed the logic and now the outbound message is cloned first and then enriched.